lib/semian.rb

```diff
@@ -156,6 +157,14 @@ def register(name, **options)
   end
 
   def retrieve_or_register(name, **args)
+    # If consumer who retrieved / registered by a Semian::Adapter, keep track
+    # of who the consumer was so that we can clear the resource reference if needed.
+    if consumer = args.delete(:consumer)
+      if consumer.class.included_modules.include?(Semian::Adapter)
```

Comment:
Actually I don't think that's correct @byroot 

```
irb(main):001:0> module Bar; end
=> nil
irb(main):002:0> class Foo
irb(main):003:1> include Bar
irb(main):004:1> end
=> Foo
irb(main):005:0> Foo.is_a?(Bar)
false
irb(main):006:0> Foo.include?(Bar)
=> true
```

So I think include? is the way to go (didn't know this shorthand)